crates/hyperqueue/src/client/commands/worker.rs

```diff
@@ -257,3 +258,42 @@ pub async fn stop_worker(session: &mut ClientSession, selector: IdSelector) -> c
 
     Ok(())
 }
+
+pub async fn wait_for_workers(
+    session: &mut ClientSession,
+    worker_count: u32,
+) -> anyhow::Result<()> {
+    async fn get_workers_status(session: &mut ClientSession) -> anyhow::Result<(u32, u32)> {
+        let msg = rpc_call!(
+            session.connection(),
+            FromClientMessage::WorkerList,
+            ToClientMessage::WorkerListResponse(r) => r
+        )
+        .await?;
+
+        let mut online: u32 = 0;
+        let mut offline: u32 = 0;
+        for info in msg.workers {
+            match info.ended {
+                None => online += 1,
+                Some(_) => offline += 1,
+            }
+        }
+        Ok((online, offline))
+    }
+
+    loop {
+        let (online, offline) = get_workers_status(session).await?;
+
+        if worker_count > online + offline {
```

Comment:
This shouldn't be an error. The command should just wait until `N` workers are online. It doesn't care about offline workers at all.